### PR TITLE
doc: [Terraform] Create new example for BYOK and multi-region cluster

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

This PR adds a new example of encryption at rest with a multi-region cluster defined as a terraform module.

Link to any related issue(s): [INTMDB-340](https://jira.mongodb.org/browse/INTMDB-340)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments
Screenshot 2023-07-26 at 16 24 38

### CODE REVIEW

File: examples/atlas-encryption-at-rest/aws/atlas-cluster/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Atlas Terraform Provider Example: Encryption at rest - AWS - Cluster
```
Comment (andreaangiolillo) [inline examples/atlas-encryption-at-rest/aws/atlas-cluster/README.md:1]: I moved the old example under the cluster folder and created the README.md

File: examples/atlas-encryption-at-rest/aws/multi-region-cluster/README.MD [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Atlas Terraform Provider Example: Encryption at rest - AWS - Multi-Region Cluster
```
Comment (andreaangiolillo) [inline examples/atlas-encryption-at-rest/aws/multi-region-cluster/README.MD:1]: new example

File: examples/atlas-encryption-at-rest/aws/multi-region-cluster/main.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
+module "multi-region-cluster" {
+  atlas_project_id   = mongodbatlas_encryption_at_rest.test.project_id
+  provider_name      = "AWS"
+  aws_region_shard_1 = var.aws_region_shard_1
+  aws_region_shard_2 = var.aws_region_shard_2
+  cluster_name       = var.cluster_name
+  instance_size      = "M10"
+  source             = "./modules/multi-region-cluster"
+}
```
Comment (andreaangiolillo) [inline examples/atlas-encryption-at-rest/aws/multi-region-cluster/main.tf:9]: the example uses a terraform module defined within the example

File: examples/atlas-api-key/create-api-key-assign-to-multiple-projects/versions.tf [hunk 4]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     mongodbatlas = {
       source  = "mongodb/mongodbatlas"
-      version = "~> 1.0"
+      version = "~> 1.10.0"
```
Comment (AgustinBettati) [inline examples/atlas-api-key/create-api-key-assign-to-multiple-projects/versions.tf:5]: Is our intention here to leave a fixed minor version? With the previous definition, any new minor version would be adopted: https://developer.hashicorp.com/terraform/language/expressions/version-constraints#-3.

**Review by AgustinBettati — CHANGES_REQUESTED** [review]

File: examples/atlas-api-key/create-api-key-assign-to-multiple-projects/versions.tf (hunk 4 above)
Comment (andreaangiolillo) [reply examples/atlas-api-key/create-api-key-assign-to-multiple-projects/versions.tf:5]: If we don't provide a version, `tflint` will complain. In this case, when running tf validate locally with `~>1.0`, terraform downloaded a provider version without the feature used in the example. Here I am adding a version that contains the feature used in the example so customers will not hit the same issue.

**Review by AgustinBettati — APPROVED** [review]

File: website/docs/r/advanced_cluster.html.markdown [hunk 5]
```diff
@@ -181,7 +181,7 @@ resource "mongodbatlas_advanced_cluster" "cluster" {
         node_count    = 1
       }
       provider_name = "GCP"
-      priority      = 0
+      priority      = 5
```
Comment (Zuhairahmed) [inline website/docs/r/advanced_cluster.html.markdown:184]: just curious, why are we changing advanced cluster priority in this doc example?

File: website/docs/r/advanced_cluster.html.markdown [hunk 6]
```diff
@@ -181,7 +181,7 @@ resource "mongodbatlas_advanced_cluster" "cluster" {
...
       region_name   = "US_EAST_4"
```
Comment (Zuhairahmed) [inline website/docs/r/advanced_cluster.html.markdown:185]: also can you confirm if we need to update priorities on cluster page as well? https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster

**Review by Zuhairahmed — APPROVED** [review]
Minor comments, not blocking. otherwise LGTM

File: website/docs/r/advanced_cluster.html.markdown (hunk 5 above)
Comment (andreaangiolillo) [reply website/docs/r/advanced_cluster.html.markdown:184]: sure, 0 is not a valid value for the priority. Getting this error

```

│ Error: error creating MongoDB ClusterAdvanced: POST https://cloud-dev.mongodb.com/api/atlas/v1.5/groups/63f4d4a47baeac59406dc131/clusters: 400 (request "INVALID_ATTRIBUTE") Invalid attribute electable region priority specified.
│
│   with mongodbatlas_advanced_cluster.cluster,
│   on main.tf line 1, in resource "mongodbatlas_advanced_cluster" "cluster":
│    1: resource "mongodbatlas_advanced_cluster" "cluster" {
```

File: website/docs/r/advanced_cluster.html.markdown (hunk 6 above)
Comment (andreaangiolillo) [reply website/docs/r/advanced_cluster.html.markdown:185]: no, the examples have the correct priority

File: website/docs/r/advanced_cluster.html.markdown [hunk 7]
```diff
@@ -48,7 +48,7 @@ resource "mongodbatlas_advanced_cluster" "test" {
         node_count    = 1
       }
       provider_name = "AWS"
-      priority      = 1
+      priority      = 7
```
Comment (andreaangiolillo) [inline website/docs/r/advanced_cluster.html.markdown:51]:
```bash
 Error: error creating MongoDB ClusterAdvanced: POST https://cloud-dev.mongodb.com/api/atlas/v1.5/groups/63f4d4a47baeac59406dc131/clusters: 400 (request "ATLAS_GENERAL_ERROR") Reason: A replication spec must specify a region with priority 7.
│
│   with mongodbatlas_advanced_cluster.test,
│   on main.tf line 60, in resource "mongodbatlas_advanced_cluster" "test":
│   60: resource "mongodbatlas_advanced_cluster" "test" {
```

**Pull request merged**